tuxemon/ai.py

```diff
@@ -81,12 +95,49 @@ def make_decision(
 class RandomAI(AI):
     """AI will use random technique against random opponent."""
 
-    def make_decision(
+    def make_decision_trainer(
+        self,
+        user: NPC,
+        monster: Monster,
+        opponents: Sequence[Monster],
+    ) -> Tuple[Union[Monster, NPC], Union[Item, Technique], Monster]:
+        """
+        Trainers battles.
+        """
+        if self.check_strongest(user, monster):
+            if len(user.inventory) > 0:
+                inventory = list(user.inventory)
+                for i in inventory:
+                    if user.is_item_sort(i, "potion"):
+                        if self.check_hp_potion(monster):

```

Comment:
I find `check_` to be a bit ambiguous as a name.

`needs_potion` ?

Reply:
Agree